src/components/icons/AnimatedIcon.tsx

```diff
@@ -16,11 +28,27 @@ const AnimatedIcon = ({
   children,
   animationComplete,
 }: AnimatedIconType) => {
+  const isLightTheme = useSelector(state => theme(state) === 'Light')
+  const lottieColor = isLightTheme ? '#000000' : '#FFFFFF'
   const newSize = size * ICON_SCALING_FACTOR
   const color = style.fill || fill || token('colors.fg')
 
+  // Create a ref to the parent div
+  const divRef = useRef<HTMLDivElement | null>(null)
+  const [dynamicColor, setDynamicColor] = useState<string | null>(null)
+
+  // Effect to calculate the color from the parent div
+  useEffect(() => {
+    if (divRef.current) {
+      const computedStyle = getComputedStyle(divRef.current)
+      const hexColor = rgbToHex(computedStyle.color) // Convert to hex
+      setDynamicColor(hexColor)
+    }
+  }, [isLightTheme, style, cssRaw]) // Dependencies that might affect the final style
```

Comment:
Is this `useEffect` needed? I think we should be able to change `dynamicColor` from being state and just declare it as a normal variable. For reference: https://react.dev/learn/you-might-not-need-an-effect#updating-state-based-on-props-or-state